cmd/jaeger/internal/extension/jaegerstorage/config.go

```diff
@@ -98,3 +104,16 @@ func (cfg *Config) Validate() error {
 	}
 	return nil
 }
+
+func (cfg *MetricBackends) Unmarshal(conf *confmap.Conf) error {
+	// apply defaults
+	if conf.IsSet("prometheus") {
+		cfg.Prometheus = &prometheus.Options{
+			Primary: prometheus.NamespaceConfig{
```

Comment:
is Options the right structure to use? The `Primary:` worries me, it seems like it was not well designed, since primary/archive were the terms used in the trace storage, but they don't really apply to Prometheus.

Maybe you can take a stab at another PR first and clean up that config. I think the only thing that's needed is `pkg/prometheus/config/Configuration`, the type `namespaceConfig` should not exist, and type Options could just embed Configuration directly (we can keep Options to minimize code changes and to isolate flags mapping functionality that is not needed for v2).

Reply:
Okay. I will create another PR for the cleanup.